deploy/basic/terraform/scripts/node.sh

```diff
@@ -34,24 +34,54 @@ MUSHOP_APP_URI=$(curl -L http://169.254.169.254/opc/v1/instance/metadata | jq -j
 WALLET_URI=$(curl -L http://169.254.169.254/opc/v1/instance/metadata | jq -j ".wallet_par")
 
 # get artifacts from object storage
-wget -O /root/wallet.zip $${WALLET_URI}
-wget -O /root/catalogue.sql $${CATALOGUE_SQL_URI}
-wget -O /etc/httpd/conf/httpd.conf $${APACHE_CONF_URI}
-wget -O /root/entrypoint.sh $${ENTRYPOINT_URI}
-
+http_status=$(curl -w '%%{http_code}' -L -s -o /root/wallet.zip $${WALLET_URI})
+if [ "$http_status" != "200" ]; then
+    echo "Retrying $${WALLET_URI}"
+    sleep 10
+    curl -L -o /root/wallet.zip $${WALLET_URI}
+fi
```

Comment:
@jeevanjoseph these seem fine, but due to repetition, I might suggest a function like the following:

```bash
# load artifact from OS with retry
# @example
#   getArtifact $CATALOGUE_SQL_URI /root/catalogue.sql
getArtifact() {
  URI=$1
  OUT=$2
  http_status=$(curl -w "%{http_code}" -L -s -o $OUT $URI)
  echo "STATUS: $http_status"
  if [ "$http_status" != "200" ]; then
      echo "Retrying $URI"
      sleep 10
      getArtifact $1 $2
  fi
}
```

Reply:
...(or adapted to only do single retry)

```bash
# load artifact from OS with retry
# @example
#   getArtifact $CATALOGUE_SQL_URI /root/catalogue.sql
getArtifact() {
  URI=$1
  OUT=$2
  http_status=$(curl -w "%{http_code}" -L -s -o $OUT $URI)
  echo "STATUS: $http_status"
  if [ "$http_status" != "200" ]; then
      echo "Retrying $URI"
      sleep 10
      curl -L -o ${OUT} ${URI}
  fi
}
```

Reply:
Let’s limit the number of retries, or would need a back off function, that would be unnecessary.

Reply:
not sure what the timeout is in the whole thing, but it might make sense for us to exit 1 after a couple of unsuccessful retries

Reply:
working on it

Reply:
PR updated